okio/src/jvmMain/kotlin/okio/Pipe.kt

```diff
@@ -32,9 +30,12 @@ import java.io.IOException
  * When the sink is closed, source reads will continue to complete normally until the buffer has
  * been exhausted. At that point reads will return -1, indicating the end of the stream. But if the
  * source is closed first, writes to the sink will immediately fail with an [IOException].
+ *
+ * A pipe may be canceled to immediately fail writes to the sink and reads from the source.
  */
 class Pipe(internal val maxBufferSize: Long) {
   internal val buffer = Buffer()
+  internal var canceled = false
```

Comment:
Should we care about any of these fields being `@Volatile`?

Reply:
No, we use `synchronized` instead.